File: tools/fix_deprecation.py
import os

files_to_fix = [
    r'frontend\node_modules\spdy\lib\spdy\agent.js',
    r'frontend\node_modules\spdy\lib\spdy\server.js',
    r'frontend\node_modules\spdy-transport\lib\spdy-transport\utils.js'
]

replacement = 'Object.assign'
target = 'util._extend'

for file_path in files_to_fix:
    full_path = os.path.join(os.getcwd(), file_path)
    if os.path.exists(full_path):
        try:
            with open(full_path, 'r', encoding='utf-8') as f:
                content = f.read()
            
            if target in content:
                new_content = content.replace(target, replacement)
                with open(full_path, 'w', encoding='utf-8') as f:
                    f.write(new_content)
                print(f"Fixed {file_path}")
            else:
                print(f"Target not found in {file_path}")
        except Exception as e:
            print(f"Error processing {file_path}: {e}")
    else:
        print(f"File not found: {full_path}")

File: tools/fix_deprecation.js
const fs = require('fs');
const path = require('path');

const filesToFix = [
    'frontend/node_modules/spdy/test/client-test.js'
];

const target = 'util._extend';
const replacement = 'Object.assign';

filesToFix.forEach(filePath => {
    const fullPath = path.join(process.cwd(), filePath);
    if (fs.existsSync(fullPath)) {
        try {
            let content = fs.readFileSync(fullPath, 'utf8');
            if (content.includes(target)) {
                content = content.split(target).join(replacement);
                fs.writeFileSync(fullPath, content, 'utf8');
                console.log(`Fixed ${filePath}`);
            } else {
                console.log(`Target not found in ${filePath}`);
            }
        } catch (e) {
            console.error(`Error processing ${filePath}:`, e);
        }
    } else {
        console.log(`File not found: ${fullPath}`);
    }
});
